grails-app/domain/org/pih/warehouse/product/Product.groovy

```diff
@@ -680,6 +681,11 @@ class Product implements Comparable, Serializable {
        return packages.collect { [uom: it.uom.code, quantity: it.quantity] }.unique()
     }
 
+    String getTranslatedNameWithLocalCode() {
+        String localeTag = LocalizationUtil.localizationService.getCurrentLocale().toLanguageTag().toUpperCase()
+        return "${name} ${translatedName ? "(${localeTag}: ${translatedName})" : ''}"
```

Comment:
You could do this as:
`"${name}${translatedName ? " (${localeTag}: ${translatedName})" : ''}"`
(remove space after `${name}` and add it before `(`, to not get dangling space at the and if there is no translation for that product).